fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBSortQueryIndexSelectionTest.java

```diff
@@ -931,4 +937,90 @@ void sortUniqueNull() {
         }
     }
 
+    /**
+     * Verify that sort that cannot be done with any index can be enabled for in-memory sort.
+     */
+    public enum SortWithoutIndexMode {
+        DISALLOWED, MEMORY, FILE
+    }
+
+    @EnumSource(SortWithoutIndexMode.class)
+    @ParameterizedTest(name = "sortWithoutIndex [mode = {0}]")
+    public void sortWithoutIndex(SortWithoutIndexMode mode) throws Exception {
```

Comment:
This test has a lot of magic numbers, coming at it from the outside, I can't easily discern whether any of the numbers are/should be correct. Perhaps adding some comments as to how they are derived, or replacing them with actual derivations would be valuable. 
For example, if you replace:
```
        if (mode == SortWithoutIndexMode.FILE) {
             executeProperties.setSkip(1001);    // Skip + limit is enough to overflow memory buffer. Skips into middle of section.
         }
```
with
```
 // Skip + limit is enough to overflow memory buffer. Skips into middle of section.
int skip = mode == SortWithoutIndexMode.FILE ? 1001 : 0;
executeProperties.setSkip(skip);
```
Would that allow later code to be something like `i = skip * 0.6` (or something like that)

Also, using the constants that exist in the production code, to derive these things would probably also make it a lot more readable.

Reply:
I rewrote it to do arithmetic on named variables, but that didn't really seem any clearer.

So I switched to a parallel iterator over (sorted) integers, formed from the same filter / skip / limit logic.